src/charm.py

```diff
@@ -730,6 +776,9 @@ def _handle_ingress_data_removed(self, event: RelationEvent):
             event.relation, wipe_rel_data=not isinstance(event, RelationBrokenEvent)
         )
 
+        if self.forward_auth.is_ready():
+            self.forward_auth.update_requirer_relation_data(self._forward_auth_config)
```

Comment:
Won't the relation still exist? This means that this call will not remove the app_name from the `forward_auth` databag. This should probably be handled by `_process_status_and_configurations`.